modules/core/depinject.go

```diff
@@ -48,27 +52,53 @@ type ModuleInputs struct {
 type ModuleOutputs struct {
 	depinject.Out
 
-	IbcKeeper *ibckeeper.Keeper
-	Module    appmodule.AppModule
+	Module appmodule.AppModule
+
+	IBCKeeper    *ibckeeper.Keeper
+	ScopedKeeper capabilitykeeper.ScopedKeeper
 }
 
+// ProvideModule defines a depinject provider function to supply the module dependencies and return its outputs.
 func ProvideModule(in ModuleInputs) ModuleOutputs {
 	// default to governance authority if not provided
 	authority := authtypes.NewModuleAddress(govtypes.ModuleName)
 	if in.Config.Authority != "" {
 		authority = authtypes.NewModuleAddressOrBech32Address(in.Config.Authority)
 	}
 
+	scopedKeeper := in.CapabilityKeeper.ScopeToModule(exported.ModuleName)
+
 	keeper := ibckeeper.NewKeeper(
 		in.Cdc,
 		in.Key,
 		in.LegacySubspace,
-		in.ConsensusHost,
+		ibctm.NewConsensusHost(in.StakingKeeper), // NOTE: need to find a way to inject a ConsensusHost into DI container created outside context of app module.
 		in.UpgradeKeeper,
-		in.ScopedKeeper,
+		scopedKeeper,
 		authority.String(),
 	)
 	m := NewAppModule(keeper)
 
-	return ModuleOutputs{IbcKeeper: keeper, Module: m}
+	return ModuleOutputs{Module: m, IBCKeeper: keeper, ScopedKeeper: scopedKeeper}
+}
+
+// InvokeAddAppRoutes defines a depinject Invoker for registering ibc application modules on the core ibc application router.
+func InvokeAddAppRoutes(keeper *ibckeeper.Keeper, appRoutes []porttypes.IBCModuleRoute) {
```

Comment:
In contrast to https://github.com/cosmos/ibc-go/pull/6216/files#r1581785698, there is a bit of a gotcha when it comes to IBC apps. We cannot simply rely on the module name as the router key as some sdk modules may container more than one `IBCModule`. For example, ica contains `icacontroller` and `icahost`. 

Here I add a kind of wrapper type to 05-port called `IBCModuleRoute`. This associated a `IBCModule` with a name key. This type implements a different depinject interface to that mentioned in https://github.com/cosmos/ibc-go/pull/6216/files#r1581785698 called `ManyPerModuleType`. We can return one or more `IBCModuleRoute` in a list from our ibc app module outputs, and as I understand the DI container is smart enough to collect these and pass to the invoker after providing each app module.